src/instance.rs

```diff
@@ -226,8 +230,39 @@ impl ChorusUser {
                 .as_ref()
                 .map(|info| info.ratelimits.clone()),
             settings,
-            object,
+            object: None,
             gateway,
         }
     }
+
+    /// Sends a request to complete an MFA challenge.
+    /// # Reference
+    /// See <https://docs.discord.sex/authentication#verify-mfa>
+    ///
+    /// If successful, the MFA verification JWT returned is set on the current [ChorusUser] executing the
+    /// request. 
+    ///
+    /// The JWT token expires after 5 minutes.
+    pub async fn complete_mfa_challenge(&mut self, mfa_verify_schema: MfaVerifySchema) -> ChorusResult<()> {
+        let endpoint_url = "/mfa/finish";
+        let chorus_request = ChorusRequest {
+            request: Client::new()
+                .post(endpoint_url)
+                .json(&mfa_verify_schema),
```

Comment:
I believe this endpoint requires authentication, you should add the user's token: `.header("Authorization", self.token())`

Reply:
I assumed that /mfa/verify would be used for all MFA verification thus a user with MFA enabled calling the login endpoint would need to call it unauthenticated, but apparently there is a specific endpoint to MFA verify login https://docs.discord.sex/authentication#verify-mfa-login.

Also my change to the user field on `ChorusUser` results from that assumption, because of the need to know if a user was authenticated or not, to define the appropriate RateLimit for the request to /mfa/verify. Not sure if you wanna keep this for other uses, although in my opinion it might be useful for other scenarios.

Reply:
> Also my change to the user field on `ChorusUser` results from that assumption, because of the need to know if a user was authenticated or not, to define the appropriate RateLimit for the request to /mfa/verify. Not sure if you wanna keep this for other uses, although in my opinion it might be useful for other scenarios.

Yeah, I think we can keep it. It's definitely better than calling `User::default()` before we're authenticated

Reply:
Fixed https://github.com/polyphony-chat/chorus/pull/521/commits/ee19cb762fbaa732f06c67cbec1f2f5dace1a80e